### PR TITLE
Introduce VolumeSnapshot APIs

### PR DESCRIPTION
**Pull request by hossainemruz**
xref: https://github.com/stashed/stash/issues/751

### CODE REVIEW

File: docs/examples/volume-snapshot/backupconfiguration.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: deployments-volume-snapshot
+  namespace: demo
+spec:
+  schedule: "* * * * *"
```
Comment (tamalsaha) [inline docs/examples/volume-snapshot/backupconfiguration.yaml]: How do you know when to use restic vs VolumeSnapshot? Something like that?
```
spec:
 driver: volumeSnapshotter
``

File: docs/examples/volume-snapshot/restoresession.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: demo-restore-from-volumesnapshot
+  namespace: demo
+spec:
+  volumeClaimTemplates:
```
Comment (tamalsaha) [inline docs/examples/volume-snapshot/restoresession.yaml]: How do you know when to use restic vs VolumeSnapshot?

Comment (hossainemruz) [reply docs/examples/volume-snapshot/restoresession.yaml]: volumeClaimTemplate will be used only for VolumeSnapshot.

Should we add driver here?

File: docs/examples/volume-snapshot/backupconfiguration.yaml (hunk 1 above)
Comment (hossainemruz) [reply docs/examples/volume-snapshot/backupconfiguration.yaml]: renamed "snapshotter" to "driver"

**Pull request merged**